lib/experimental/block-comments.php

```diff
@@ -93,3 +93,20 @@ function exclude_block_comments_from_admin( $clauses, $query ) {
 	}
 	add_action( 'comments_clauses', 'exclude_block_comments_from_admin', 10, 2 );
 }
+
+/**
+ * Enable empty comments when the comment_type is 'block_comment_reopened' or 'block_comment_resolved'.
+ *
+ * @param bool  $allow            Whether to allow an empty comment.
+ * @param array $prepared_comment The prepared comment data.
+ * @return bool Modified allow empty comment value.
+ */
+function gutenberg_allow_empty_block_comments( $allow, $prepared_comment ) {
+	if ( 'block_comment_reopened' === $prepared_comment['comment_type'] ||
+		'block_comment_resolved' === $prepared_comment['comment_type'] ) {
+		$allow = true;
+	}
```

Comment:
```suggestion
	if (
		isset( $prepared_comment['comment_type'] ) &&
		'block_comment_reopened' === $prepared_comment['comment_type'] ||
		'block_comment_resolved' === $prepared_comment['comment_type']
	) {
		$allow = true;
	}
```

Otherwise, the unit test fails with the following message:

```
1) Gutenberg_Allow_Empty_Block_Comments_Test::test_handles_missing_comment_type
Undefined array key "comment_type"

/var/www/html/wp-content/plugins/gutenberg/lib/experimental/block-comments.php:105
/var/www/html/wp-content/plugins/gutenberg/phpunit/experimental/block-comments-test.php:105
```